src/main/drivers/accgyro_mpu.c

```diff
@@ -0,0 +1,368 @@
+        mpuConfiguration.write = mpu6000WriteRegister;
+        return true;
+    }
+#endif
+
+    return false;
+}
+#endif
+
+static void mpu6050FindRevision(void)
+{
+    bool ack;
+    UNUSED(ack);
+
+    uint8_t readBuffer[6];
+    uint8_t revision;
+    uint8_t productId;
+
+    // There is a map of revision contained in the android source tree which is quite comprehensive and may help to understand this code
+    // See https://android.googlesource.com/kernel/msm.git/+/eaf36994a3992b8f918c18e4f7411e8b2320a35f/drivers/misc/mpu6050/mldl_cfg.c
+
+    // determine product ID and accel revision
+    ack = i2cRead(MPU_ADDRESS, MPU_RA_XA_OFFS_H, 6, readBuffer);
+    revision = ((readBuffer[5] & 0x01) << 2) | ((readBuffer[3] & 0x01) << 1) | (readBuffer[1] & 0x01);
+    if (revision) {
+        /* Congrats, these parts are better. */
+        if (revision == 1) {
+            mpuDetectionResult.resolution = MPU_HALF_RESOLUTION;
+        } else if (revision == 2) {
+            mpuDetectionResult.resolution = MPU_FULL_RESOLUTION;
+        } else {
+            failureMode(FAILURE_ACC_INCOMPATIBLE);
+        }
+    } else {
+        ack = i2cRead(MPU_ADDRESS, MPU_RA_PRODUCT_ID, 1, &productId);
+        revision = productId & 0x0F;
+        if (!revision) {
+            failureMode(FAILURE_ACC_INCOMPATIBLE);
+        } else if (revision == 4) {
+            mpuDetectionResult.resolution = MPU_HALF_RESOLUTION;
+        } else {
+            mpuDetectionResult.resolution = MPU_FULL_RESOLUTION;
+        }
+    }
+}
+
+void MPU_DATA_READY_EXTI_Handler(void)
+{
+    if (EXTI_GetITStatus(mpuIntExtiConfig->exti_line) == RESET) {
+        return;
+    }
+
+    EXTI_ClearITPendingBit(mpuIntExtiConfig->exti_line);
+
+#ifdef DEBUG_MPU_DATA_READY_INTERRUPT
+    // Measure the delta in micro seconds between calls to the interrupt handler
+    static uint32_t lastCalledAt = 0;
+    static int32_t callDelta = 0;
+
+    uint32_t now = micros();
+    callDelta = now - lastCalledAt;
+
+    //UNUSED(callDelta);
+    debug[0] = callDelta;
+
+    lastCalledAt = now;
+#endif
+}
+
+void configureMPUDataReadyInterruptHandling(void)
+{
+#ifdef USE_MPU_DATA_READY_SIGNAL
+
+#ifdef STM32F10X
+    // enable AFIO for EXTI support
+    RCC_APB2PeriphClockCmd(RCC_APB2Periph_AFIO, ENABLE);
+#endif
+
+#ifdef STM32F303xC
+    /* Enable SYSCFG clock otherwise the EXTI irq handlers are not called */
+    RCC_APB2PeriphClockCmd(RCC_APB2Periph_SYSCFG, ENABLE);
+#endif
+
+#ifdef STM32F10X
+    gpioExtiLineConfig(mpuIntExtiConfig->exti_port_source, mpuIntExtiConfig->exti_pin_source);
+#endif
+
+#ifdef STM32F303xC
+    gpioExtiLineConfig(mpuIntExtiConfig->exti_port_source, mpuIntExtiConfig->exti_pin_source);
+#endif
+
+#ifdef ENSURE_MPU_DATA_READY_IS_LOW
+    uint8_t status = GPIO_ReadInputDataBit(mpuIntExtiConfig->gpioPort, mpuIntExtiConfig->gpioPin);
+    if (status) {
+        return;
+    }
+#endif
+
+    registerExti15_10_CallbackHandler(MPU_DATA_READY_EXTI_Handler);
```

Comment:
This will cause problems with interrupts on other pins than 10-15. 
I faced this problem yesterday when i implemented the interrupt for the cc3d here : https://github.com/henn1001/cleanflight/commits/betaflight

The handler is here: https://github.com/cleanflight/cleanflight/blob/master/src/main/drivers/system.c#L39

As you can see i did a workaround and hardcoded the interrupt for pin 3 in mpu6000 but this is obviously not the right solution.